packages/truffle/truffle-config.js

```diff
@@ -1,15 +1,259 @@
   // for more details on how to specify configuration options!
   //
+  /**
+  * contracts_build_directory tells Truffle where to store compiled contracts
+  */
+  contracts_build_directory: '../react-app/src/contracts/',
   networks: {
     development: {
-      host: "127.0.0.1",
-      port: 8545,
+      url: "http://localhost:8545",
+      network_id: "*"
+      /*
+        we set a mnemonic to use with our ganache node, and can deploy through it
+
+      */
+    },
+    localhost: {
+      url: "http://localhost:8545",
       network_id: "*"
+      /*
+        we set a mnemonic to use with our ganache node, and can deploy through it
+
+      */
+    },
+
+    // rinkeby: {
+    //   url: `https://rinkeby.infura.io/v3/${process.env.RINKEBY_INFURA_KEY}`,
+    //   accounts: [`${process.env.RINKEBY_DEPLOYER_PRIV_KEY}`],
+    // },
+    // kovan: {
+    //   url: `https://rinkeby.infura.io/v3/${process.env.KOVAN_INFURA_KEY}`,
+    //   accounts: [`${process.env.KOVAN_DEPLOYER_PRIV_KEY}`],
+    // },
+    // mainnet: {
+    //   url: `https://mainnet.infura.io/v3/${process.env.MAINNET_INFURA_KEY}`,
+    //   accounts: [`${process.env.MAINNET_DEPLOYER_PRIV_KEY}`],
+    // },
+    // ropsten: {
+    //   url: `https://ropsten.infura.io/v3/${process.env.ROPSTEN_INFURA_KEY}`,
+    //   accounts: [`${process.env.ROPSTEN_DEPLOYER_PRIV_KEY}`],
+    // },
+    // goerli: {
+    //   url: `https://goerli.infura.io/v3/${process.env.GOERLI_INFURA_KEY}`,
+    //   accounts: [`${process.env.GOERLI_DEPLOYER_PRIV_KEY}`],
+    // },
+    // xdai: {
+    //   url: 'https://dai.poa.network',
+    //   gasPrice: 1000000000,
+    //   accounts: [`${process.env.XDAI_DEPLOYER_PRIV_KEY}`],
+    // },
+
+    rinkeby: {
+      url: "https://rinkeby.infura.io/v3/460f40a260564ac4a4f4b3fffb032dad", // <---- YOUR INFURA ID! (or it won't work)
+
+      //    url: "https://speedy-nodes-nyc.moralis.io/XXXXXXXXXXXXXXXXXXXXXXX/eth/rinkeby", // <---- YOUR MORALIS ID! (not limited to infura)
+
+      accounts: {
+        mnemonic: mnemonic(),
+      },
+    },
+    kovan: {
+      url: "https://kovan.infura.io/v3/460f40a260564ac4a4f4b3fffb032dad", // <---- YOUR INFURA ID! (or it won't work)
+
+      //    url: "https://speedy-nodes-nyc.moralis.io/XXXXXXXXXXXXXXXXXXXXXXX/eth/kovan", // <---- YOUR MORALIS ID! (not limited to infura)
+
+      accounts: {
+        mnemonic: mnemonic(),
+      },
+    },
+    mainnet: {
+      url: "https://mainnet.infura.io/v3/460f40a260564ac4a4f4b3fffb032dad", // <---- YOUR INFURA ID! (or it won't work)
+
+      //      url: "https://speedy-nodes-nyc.moralis.io/XXXXXXXXXXXXXXXXXXXXXXXXX/eth/mainnet", // <---- YOUR MORALIS ID! (not limited to infura)
+
+      gasPrice: mainnetGwei * 1000000000,
+      accounts: {
+        mnemonic: mnemonic(),
+      },
+    },
+    ropsten: {
+      url: "https://ropsten.infura.io/v3/460f40a260564ac4a4f4b3fffb032dad", // <---- YOUR INFURA ID! (or it won't work)
```

Comment:
this one as well, not sure if it needs to be here or not but may have been left on accident.